src/backend/distributed/utils/metadata_cache.c

```diff
@@ -1181,13 +1180,22 @@ CheckCitusVersion(int elevel)
  * this function logs an error with the specified elevel and returns false,
  * otherwise it returns true.
  */
-static bool
+bool
 CheckAvailableVersion(int elevel)
 {
 	char *availableVersion = NULL;
 
-	Assert(CitusHasBeenLoaded());
-	Assert(EnableVersionChecks);
+	/*
+	 * Normally CitusHasBeenLoaded() would return false because creating_extension
+	 * is set to true during ALTER EXTENSION ... UPDATE by PostgreSQL. However we hit
+	 * this function before PostgreSQL processes ALTER EXTENSION citus UPDATE. Therefore
+	 * we will not incorrectly return early because of the check below.
```

Comment:
This comment is a bit hard to follow because `CitusHasBeenLoaded` doesn't get called anymore. I think we can just omit it now.